java/ql/src/Security/CWE/CWE-297/UnsafeHostnameVerification.qhelp

```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>
+If a <code>HostnameVerifier</code> always returns <code>true</code> it will not verify the hostname at all.
+This stops Transport Layer Security (TLS) providing any security and allows an attacker to perform a man-in-the-middle attack against the application.
+</p>
+
+<p>
+An attack might look like this:
+</p>
+
+<ol>
+  <li>The program connects to <code>https://example.com</code>.</li>
+  <li>The attacker intercepts this connection and presents an apparently-valid certificate of their choosing.</li>
+  <li>The `TrustManager` of the program verifies that the certificate has been issued by a trusted certificate authority.</li>
+  <li>Java checks whether the certificate has been issued for the host <code>example.com</code>. This check fails because the certificate has been issued for a domain controlled by the attacker, for example: <code>malicious.domain</code>.</li>
+  <li>Java wants to reject the certificate because the hostname does not match. Before doing this it checks whether a <code>HostnameVerifier</code> exists.</li>
```

Comment:
Thanks for the update to line 18 - "TrustManager" sounds like a good solution. 

Would it make sense to revise lines 19, 20, and 22 in this way too?

Also, it looks as if you intended "TrustManager" to be shown in a monospaced font, in which case you'll need to use `<code>` tags.

Reply:
I've revised lines 18, 19, 20 and 22.
Hope this is fine now.

I also fixed the monospace font issue by using the right tags^^